ts/src/vertex.ts

```diff
@@ -0,0 +1,1741 @@
+            // 1 << 62 = 4611686018427387904
+            expiration = Precise.stringOr (expiration, '4611686018427387904');
+        } else if (timeInForce === 'fok') {
+            // 2 << 62 = 9223372036854775808
+            expiration = Precise.stringOr (expiration, '9223372036854775808');
+        } else if (postOnly) {
+            // 3 << 62 = 13835058055282163712
+            expiration = Precise.stringOr (expiration, '13835058055282163712');
+            if (reduceOnly) {
+                throw new NotSupported (this.id + ' reduceOnly not supported when postOnly is enabled' );
+            }
+        }
+        if (isTrigger) {
+            // 1 << 63 = 9223372036854775808
+            nonce = Precise.stringOr (nonce, '9223372036854775808');
+        }
+        if (reduceOnly) {
+            // 1 << 61 = 2305843009213693952
+            expiration = Precise.stringOr (expiration, '2305843009213693952');
+        }
+        const order = {
+            'sender': this.convertAddressToSender (this.walletAddress),
+            'nonce': nonce,
+            'expiration': expiration,
+            'priceX18': this.convertToX18 (this.priceToPrecision (symbol, price)),
+        };
+        if (side === 'sell') {
+            if (amount > 0) {
+                amount *= -1;
+            }
+        } else {
+            if (amount < 0) {
+                amount *= -1;
+            }
+        }
+        order['amount'] = this.convertToX18 (this.amountToPrecision (symbol, amount));
+        const request = {
+            'place_order': {
+                'product_id': marketId,
+                'order': {
+                    'sender': order['sender'],
+                    'nonce': order['nonce'],
+                    'expiration': this.numberToString (order['expiration']),
+                    'priceX18': order['priceX18'],
+                    'amount': order['amount'],
+                },
+                'signature': this.buildCreateOrderSig (order, chainId, verifyingContractAddress),
+            },
+        };
+        params = this.omit (params, [ 'timeInForce', 'reduceOnly', 'postOnly', 'triggerPrice', 'stopPrice', 'stopLossPrice', 'takeProfitPrice' ]);
+        let response = undefined;
+        if (isTrigger) {
+            const trigger = {};
+            if (stopLossPrice !== undefined) {
+                trigger['last_price_below'] = this.convertToX18 (stopLossPrice);
+            } else if (takeProfitPrice !== undefined) {
+                trigger['last_price_above'] = this.convertToX18 (takeProfitPrice);
+            }
+            request['place_order']['trigger'] = trigger;
+            response = await this.v1TriggerPostExecute (this.extend (request, params));
+        } else {
+            response = await this.v1GatewayPostExecute (this.extend (request, params));
+        }
+        //
+        // {
+        //     "status": "success",
+        //     "signature": {signature},
+        //     "data": { 
```

Comment:
there are a lot of trailing whitespaces 😅 
![image](https://github.com/ccxt/ccxt/assets/43336371/5ccaa6ed-b894-41c7-a4cf-d1026f8723bc)